### DISCUSSION TITLE
Multi-Keyboard Layout Support

### DISCUSSION THREAD

**Discussion by euvitech** [mulaRahul/keyviz#16] [Ideas]
# keyboard alternative

I live in Brazil and the **organization of my keyboard** is different from the US, which makes some things difficult when typing.

It would be very interesting if the program could recognize which **keyboard style** the person is using.

An alternative solution is also for the person to configure the keyboard by the program, as some keyboards have the *"ç"* and others do not, as well as some have no other functionality.

**Comment by mulaRahul** [comment]
As the project is in its early stages, only **English (US)** keyboard layout is supported. In the upcoming releases, support for other keyboard layouts will be added. Though, I can't guarantee that this feature will be worked on in the near future.

**Reply by euvitech** [reply]
Got it, I really liked your program, it helps me a lot, especially when I'm going to teach someone a shortcut

**Reply by BannukDE** [reply]
I think we as a community could do some of the work by collecting the necessary data.
qwertz_de here!
Similar to a multilingual table, the data could be collected together based on the English keyboard at first, maybe there is already something similar on the internet... as soon as I find some time I will investigate this.

**Reply by mulaRahul** [reply]
@BannukDE any help is much appreciated! Currently, the application relies on another [dependency](https://github.com/Krysl/keyboard_event) for the keyboard inputs (that only supports windows), and _kinda_ captures a lot of things like normal keys, volume up, _kanji_, etc. I'm planning to create a custom keyboard-input listener to extend the application for other platforms (Mac, Linux) and keyboard layouts.

**Comment by marckl67** [comment]
I already worked on a French AZERTY keyboard layout on PC. I will be happy to help working on extending the keyboards layouts.

**Comment by marc-romu** [comment]
Hi @mulaRahul ! Please check PR #494, which enables Keyviz to be extended to any keyboard layout. I've initially added Spanish (Spain) QWERTY (the layout I can test). I propose an extensible system where adding a new layout requires only one extra file.

**Reply by InSasquatchCountry** [reply]
As a colemak user, I would love this.

**Reply by js485** [reply]
Would be great to have this and a layout file for German QWERTZ in addition. :)

**Reply by marc-romu** [reply]
Hi all! I compiled my PR, which includes support for multiple layouts, and published it here:
https://github.com/marc-romu/keyviz/releases/tag/2.1.2-dev.260822

I included layouts for German, French, Colemak and Spanish (Spain).

It is pretty easy to add new layouts now, so feel free to ask for more or contribute yourself with new ones.

**Reply by InSasquatchCountry** [reply]
@marc-romu I have moved to [Keyty](https://github.com/keytyapp/Keyty/) 

👍 10/10 would recommend 🙌

2026-08-23 edit: The new release V1.4 just dropped. The dev polished a lot stuff and added new features.
